pretyping/nativenorm.ml

```diff
@@ -399,22 +400,31 @@ and  nf_predicate env sigma ind mip params v pT =
       mkLambda(name,dom,body)
     | _ -> nf_type env sigma v
 
-and nf_evar env sigma evk ty args =
+and nf_evar env sigma evk args =
   let evi = try Evd.find sigma evk with Not_found -> assert false in
   let hyps = Environ.named_context_of_val (Evd.evar_filtered_hyps evi) in
+  let ty = EConstr.Unsafe.to_constr @@ Evd.evar_concl evi in
   if List.is_empty hyps then begin
     assert (Int.equal (Array.length args) 0);
     mkEvar (evk, [||]), ty
   end
   else
+    let drop_let decl = match decl with
```

Comment:
Same remark as the other PR.

Reply:
OK, I introduced the combinator in the other PR, so that I'll rebase that one atop of the other once it is merged.